### DISCUSSION TITLE
How to use Shifu as a tcp server

### DISCUSSION THREAD

**Discussion by codering** [Edgenesis/shifu#887] [Q&A]
After reviewing the Shifu documentation, I noticed that Shifu typically establishes connections with physical devices and retrieves data from them. However, if my device operates using the TCP protocol and needs to connect to a TCP server, automatically transmitting data to the TCP server upon receiving certain commands, with the TCP server performing data transformation and forwarding it to an MQTT broker, can Shifu be used as the TCP server to handle this situation?

**Comment by tomqin93** [comment]
Hi @codering , apologies for the late reply.
Regarding your question, currently Shifu supports continuous streaming TCP (example configuration here: https://github.com/Edgenesis/shifu/tree/main/examples/tcpDeviceShifu) but not to MQTT. Could you please elaborate on what the data is like from TCP to MQTT? Feel free to add a feature request for this. Thanks!!
